### PR TITLE
GitLab related doc update

### PR DESCRIPTION
**Pull request by guineveresaenger**
…tions, pending ultimate synching of gitlab to github

Added instructions for setting up gitlab with solas-derived repositories.

### CODE REVIEW

**Comment by leahnp** [comment]
@guineveresaenger This is looking great - nice work! I think we could merge a first iteration, are you comfortable merging considering the outstanding `ToDo` section?

Also I'd like to follow the instructions on a real port - do we have a master ticket that lists the repos we need to add to GitLab?

**Comment by guineveresaenger** [comment]
@leahnp  I would like to wait until we have a cleanup stage added to the template. Hopefully, that wouldn't need extra manual changes from the ones outlined in the directions, but if it does, I'd rather change it on an open PR.

**Comment by guineveresaenger** [comment]
This is getting closer to getting done and might even warrant a preliminary merge. All about-to-be deprecated instructions are clearly labeled as such. 
One point of interest is whether we want to continue using a Kubernetes Secret for the robot password. GitLab Secrets are pretty easy to use, but I am not sure about the ramifications from a security standpoint.

File: docs/gitlab.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+* Define Gitlab secrets
+
+  * Head to your solas repo on Gitlab. Go to `Settings` --> `CI/CD` and expand `Secret Variables`.
+  ![screenshot](images/gitlab/gitlab-settings.png)
+  * Create a `QUAY_ROBOT` Secret Key, and assign it the name of the robot created during the [Quay](./quay.md) configuration. 
+  * Similarly, create a `QUAY_PASSWORD` Secret Key and assign it the quay robot's docker login password.
+  ![screenshot](images/gitlab/gitlab-secrets.png)
+  * For container repositories:
+```
+def QUAY_ROBOT            = "zabra-container";
```
[the remark this answers is not in the document]
Comment (guineveresaenger) [reply docs/gitlab.md]: It's what it came with, and I tend to agree on it being confusing, so I'll take it out.

File: docs/gitlab.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+### GitLab environment variables
+GitLab comes with a list of handy built-in environment variables, some of which are used in the CI file. 
+[Reference here](http://docs.gitlab.com/ce/ci/variables/README.html#predefined-variables-environment-variables)
+
+### Define GitLab secrets
+
+  * Head to your solas repo on Gitlab. Go to `Settings` --> `CI/CD` and expand `Secret Variables`.
+  ![screenshot](images/gitlab/gitlab-settings.png)
+  * Create a `QUAY_ROBOT` Secret Key, and assign it the name of the robot created during the [Quay](./quay.md) configuration.
```
Comment (guineveresaenger) [inline docs/gitlab.md]: reviewing my own markdown: these links are broken

File: docs/gitlab.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+### Optional Cleanup stage
+
+Some repositories build very large images, and we may want to clean up any unused branch builds on the GitLab docker image repository. This is currently an upstream work in progress; in the meantime if necessary you can implement it yourself.
+
+First, for a sample cleanup stage, please see the cleanup stage of [container-fluent-bit](https://github.com/samsung-cnct/container-fluent-bit/blob/master/.gitlab-ci.yml). You should be able to copy this verbatim, and then provide some GitLab configurations to make it work:
+
+1. Go to User -> Settings and request a [personal access token] (https://docs.gitlab.com/ce/user/profile/personal_access_tokens.html), with API access (read and write)
+2. In your Gitlab project repository, follow the steps for [GitLab secrets](#Define GitLab secrets) to create the following additional secrets:
```
Comment (guineveresaenger) [inline docs/gitlab.md]: reviewing my own markdown failures: these links are broken

**Comment by guineveresaenger** [comment]
Should be all better now.

**Pull request merged**